x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/multiterms/MultiTermsAggregator.java

```diff
@@ -95,11 +95,14 @@ protected MultiTermsAggregator(
         partiallyBuiltBucketComparator = order == null ? null : order.partiallyBuiltBucketComparator(b -> b.bucketOrd, this);
         this.formats = formats;
         this.showTermDocCountError = showTermDocCountError;
-        if (subAggsNeedScore() && descendsFromNestedAggregator(parent)) {
+        if (subAggsNeedScore() && descendsFromNestedAggregator(parent) || context.isInSortOrderExecutionRequired()) {
```

Comment:
For time series indices `isInSortOrderExecutionRequired` = `true` which means that the boolean condition will be:
`subAggsNeedScore() AND descendsFromNestedAggregator(parent) OR true` which will always have result `true` no matter the value of `subAggsNeedScore()` and `descendsFromNestedAggregator(parent)`. Note that `&&` has also precedence on `||`.

I expect instead that if `subAggsNeedScore() && descendsFromNestedAggregator(parent)` == `false` this is not executed (as it was before introducing the additional OR)...while as it is now it will be executed in case `context.isInSortOrderExecutionRequired()` == `true`

This means we have
```
false && false || false = false (OK)
false && false || true = true -> while it should be false, right?
false && true || true = true -> while it should be false, right?
false && true || false = true -> while it should be false, right?
true && false || true = -> while it should be false, right?
true && false || false = false (OK)
true && true || true = true (OK)
true && true || false = true (OK)
```